t/db_dependent/api/v1/holdings.t

```diff
@@ -117,9 +117,75 @@ subtest 'get() tests' => sub {
 };
 
 subtest 'add() tests' => sub {
-    plan skip_all => 'not implemented';
+    plan tests => 8;
 
     $schema->storage->txn_begin;
+
+    my ($patron, $session_patron) = create_user_and_session( { authorized => 0 } );
+    my ($librarian, $session_librarian) = create_user_and_session( { authorized => 1 } );
+
+    my $branchcode1 = $builder->build( { source => 'Branch' } )->{branchcode};
+    my $branchcode2 = $builder->build( { source => 'Branch' } )->{branchcode};
+    create_framework();
+
+    my $holding_marc = MARC::Record->new();
+    $holding_marc->append_fields(MARC::Field->new(
+        '852','','','b' => $branchcode1, 'c' => $branchcode2));
+
+    Koha::Holdings->search->delete;
+
+    my $biblionumber = create_biblio( 'test' );
+    my $biblio = Koha::Biblios->find( $biblionumber );
+
+    my $holding = {
+        biblionumber     => $biblionumber,
+        biblioitemnumber => $biblio->biblioitem->biblioitemnumber,
+        holdingbranch    => $branchcode1,
+        location         => $branchcode2,
+        frameworkcode    => 'HLD',
+        metadata         => {
+            metadata => $holding_marc->as_xml()
+        }
+    };
+
+    $tx = $t->ua->build_tx( POST => '/api/v1/holdings' => json => $holding );
+    $t->request_ok( $tx )->status_is( 401 );
+
+    $tx = $t->ua->build_tx( POST => '/api/v1/holdings' => json => $holding );
+    $tx->req->cookies( { name => 'CGISESSID', value => $session_patron } );
+    $t->request_ok( $tx )->status_is( 403 );
+
+    subtest 'test required parameters' => sub {
+        plan tests => 9;
+
+        delete $holding->{biblionumber};
+        $tx = $t->ua->build_tx( POST => '/api/v1/holdings/' => json => $holding );
+        $tx->req->cookies( { name => 'CGISESSID', value => $session_librarian } );
+        $t->request_ok( $tx )->status_is( 400 )
+          ->json_like('/errors/0/path' => qr/biblionumber/);
+        $holding->{biblionumber} = $biblionumber;
+
+        delete $holding->{biblioitemnumber};
+        $tx = $t->ua->build_tx( POST => '/api/v1/holdings/' => json => $holding );
+        $tx->req->cookies( { name => 'CGISESSID', value => $session_librarian } );
+        $t->request_ok( $tx )->status_is( 400 )
+          ->json_like('/errors/0/path' => qr/biblioitemnumber/);
+        $holding->{biblioitemnumber} = $biblio->biblioitem->biblioitemnumber;
```

Comment:
I cant believe this is happening ;)

Reply:
What do you suggest? $biblio->biblioitemnumber? The schema does not yet support it and implementing support for it is out of scope of this bug.